### ISSUE TITLE
Drag of JavaScript Chart Editable Annotations at the time of insertion

### ISSUE THREAD

**Issue by sakshamjaiswal1** [ABTSoftware/SciChart.JS.Examples#181]
We are using JavaScript Chart Editable Annotations but we need some more modificaions in that which can enhance this feature in terms of user experience.
Currently when the anonations are loaded they are havinng a fixed shape but we want the annotations to be draggable as soon as they are loaded in the charts whitout any extra clicks,this feature is provided by tradingView charts and we also want to provide the same to our users.
I have attached images and videos for better understanding of the issue.
Hope we can get the solution of this issue asap.

Please refer to the below video link for beter understanding.
https://www.scichart.com/wp-content/sabai/File/files/l_58835d6c84c3ee425515c6f130db83d3.jpg

**Comment by andyb1979** [comment]
Hi there, is this the same requirement as [this forum question](https://www.scichart.com/questions/js/drag-of-javascript-chart-editable-annotations-at-the-time-of-insertion)?

If so, we don't support this right now, but David (@antichaosdb) our CTO thinks it should be possible via a custom modifier. He was looking at it recently. David - any thoughts?

**Comment by andyb1979** [comment]
Actually, the [forum question](https://www.scichart.com/questions/js/drag-of-javascript-chart-editable-annotations-at-the-time-of-insertion) I linked to has David's answer. Can you check it?

Curious why you didn't receive an email notification of updated answer

**Comment by sakshamjaiswal1** [comment]
hey i have just now checked the answer but did not able to get few of the concepts can i get a working example so that i can implement it and run in my app it will be very helpfull if i can get codepen working demo
@andyb1979 Thnks in advance

**Comment by andyb1979** [comment]
understand, David did mention in the [forum post](https://www.scichart.com/questions/js/drag-of-javascript-chart-editable-annotations-at-the-time-of-insertion) that he plans to make a full example at some point however we have other priorities at the moment.

My advice for now is take the example we posted in [github here](https://github.com/ABTSoftware/SciChart.JS.Examples/tree/master/Sandbox/CustomerExamples/CreateAnnotationsDynamically). Substitute the [CreateAnnotationModifier here](https://www.scichart.com/questions/js/drag-of-javascript-chart-editable-annotations-at-the-time-of-insertion). Does this show the concept? 

Basically what we're doing is overriding our ChartModifierBase API to allow you to control precisely what interactions happen on mouse-down and how to place annotations + edit them immediately.
